roles/addons/diskless/files/disklessset.py

```diff
@@ -356,7 +397,13 @@ def generate_ipxe_boot_file(image_type, image_name, image_kernel, image_initramf
             elif selected_livenet_type == '1':
                 os.system('dnf groupinstall -y "core"  --setopt=module_platform_id=platform:el8 --installroot=/mnt')
             elif selected_livenet_type == '4':
-                os.system('dnf install -y @core {0} --setopt=module_platform_id=platform:el8 --installroot=/mnt'.format(selected_packages_list))
+                try:
+                    if os.system('dnf install -y @core {0} --setopt=module_platform_id=platform:el8 --installroot=/mnt'.format(selected_packages_list)) != 0:
+                        raise Exception('dnf does not exist')
```

Comment:
dnf can exist but exit with rc!=0

    dnf(8):
       Return values:
       · 0  : Operation was successful.
       · 1  : An error occurred, which was handled by dnf.
       · 3  : An unknown unhandled error occurred during operation.
       · 100: See check-update
       · 200: There was a problem with acquiring or releasing of locks.